subsys/net/ip/connection.c

```diff
@@ -827,7 +827,9 @@ enum net_verdict net_conn_input(struct net_pkt *pkt,
 		    net_pkt_family(pkt) == AF_PACKET) {
 		;
 	} else {
+#if !CONFIG_NET_STEALTH_UNREACHABLE_DESTINATION
```

Comment:
For this particular case, it'd be better to provide a dummy implementation ofr `conn_send_icmp_error()` instead of ifdef slicing the code, it's a common practice accross the net stack, see https://github.com/zephyrproject-rtos/zephyr/blob/fe2bc27df3b51b55bf6c69a72d3ff0b1f377ca67/subsys/net/ip/connection.c#L113 for example.

And please use `if !defined(CONFIG_NET_STEALTH_UNREACHABLE_DESTINATION)` instead.

Reply:
Done.